diff --git a/server/world_router.go b/server/world_router.go
index e89d3f2c..22806168 100644
--- a/server/world_router.go
+++ b/server/world_router.go
@@ -5,7 +5,9 @@ import (
 	"server/server/timing"
 
 	credis "github.com/go-redis/redis/v8"
+	"go.uber.org/zap"
 	"golang.org/x/net/context"
+	"sync"
 )
 var(
 
@@ -46,4 +48,197 @@ func NewWorld4Component(name string, dburl, rurl string) *WorldComp {
 
 func RegisterWorldRouter(mode SceneWorldMode, cb FuncWorldRoute){
     world_router_map[mode] = cb
-}
\ No newline at end of file
+}
+
+/////////////////////////////////////
+func RegisterDoWorldTask(mode int, cb func(string)IDoWorldTask){
+
+    world_task_process[mode] = cb
+}
+
+
+func GSceneWorldRouter(name string, dburl, rurl string)*WorldComp {
+    cfg := NewGameSceneConfig(name)
+    if !cfg.IsGScene(){
+        return NewWorld4Component(name, dburl, rurl)
+    }
+    sid := GetServerIdByGId(TODOCTX, SysLogger, GuildIdType(cfg.SceneId()))
+    if sid != TryGetServerId(){
+        return NewWorld4Component(name, dburl, rurl)
+    }
+    cc := ServerWorld.cache()
+    cache, ok := cc.(*storagetask.IBaseCachePool)
+    if !ok{
+        return NewWorld4Component(name, dburl, rurl)
+    }
+    z := &WorldComp{
+    	db:     ServerWorld.db().Clone(),
+    	c:      cache.Clone(),
+    	man:    ServerWorld.man().Clone(),
+    	timing: ServerWorld.timing().Clone(),
+        unreal: true,
+    }
+    return z
+}
+func GSceneDoWorldTask(name string)IDoWorldTask{
+	return NewBaseSceneDoWorldTask(name, 2, 4)
+
+}
+
+var(
+
+	newbie_dotask = &NewBieDoScenWorldTask{
+		CDoScenWorldTask: _NewBaseSceneDoWorldTask("NewBie", 32, 1),
+	}
+)
+func NewBieSceneDoWorldTask(name string)IDoWorldTask{
+	newbie_dotask.Run()
+	return newbie_dotask
+}
+
+
+type NewBieDoScenWorldTask struct {
+	*CDoScenWorldTask
+	total int
+	runOnce sync.Once
+
+}
+func (s *NewBieDoScenWorldTask)Init(){
+	cb := func(){
+		s.Run()
+	}
+	s.runOnce.Do(cb)
+
+}
+func (s *NewBieDoScenWorldTask) Stop(){
+
+
+}
+func (s *NewBieDoScenWorldTask) Start(){
+
+
+}
+
+func (s *NewBieDoScenWorldTask) DoTask(task worldTask){
+	sz := len(s.taskCh)
+	if sz <= 0{
+		return
+	}
+	s.total += 1
+    idx := s.total % sz
+    if !s.state[idx]{
+        return
+    }
+    s.taskCh[idx] <- task
+
+}
+
+func NewBaseSceneDoWorldTask(name string, count int, queue int)IDoWorldTask{
+	return _NewBaseSceneDoWorldTask(name, count, queue)
+}
+
+func _NewBaseSceneDoWorldTask(name string, count int, queue int)*CDoScenWorldTask{
+    inst := &CDoScenWorldTask{
+		Name: name,
+    	_count: count,
+    	_run:  false,
+    	taskCh: make([]chan worldTask, count),
+    	state:  make([]bool, count),
+    }
+    for i := range inst.taskCh {
+        inst.taskCh[i] = make(chan worldTask, queue)
+		inst.state[i] = false
+    }
+    
+    return inst
+}
+
+
+type CDoScenWorldTask struct {
+	Name string
+	_count int
+    _run bool
+	taskCh     []chan worldTask
+	state     []bool
+}
+
+func (s *CDoScenWorldTask)Start(){
+	s.Run()
+
+}
+
+func (s *CDoScenWorldTask) DoTask(task worldTask){
+	sz := len(s.taskCh)
+	if sz <= 0{
+		return
+	}
+	pos := task.Pos
+    idx := pos.X % sz
+    if !s.state[idx]{
+        return
+    }
+    s.taskCh[idx] <- task
+
+}
+func (s *CDoScenWorldTask)Stop(){
+	if !s._run{
+		return
+	}
+	s._run = false
+   	{
+        for i := range s.taskCh {
+			if !s.state[i]{
+				continue
+			}
+            s.state[i] = false
+            drain:
+            for {
+                select {
+                case task, ok := <- s.taskCh[i]:
+                    if ok {
+                        ServerWorld.SaveTask(task.W.Name(), task.Msg)
+                    }
+                default:
+                    break drain
+                }
+            }
+            close(s.taskCh[i])
+        }
+    }
+}
+func (s *CDoScenWorldTask) Run() {
+	if s._run{
+		return
+	}
+	s._run = true
+    for i := range s.taskCh {
+		if s.state[i]{
+			continue
+		}
+        s.state[i] = true
+        go func(idx int) {
+			defer HandleCrash(SysLogger, "CDoScenWorldTask:"+ s.Name)
+            for task := range s.taskCh[idx] {
+				func(){
+					defer HandleCrash(SysLogger, "CDoScenWorldTaskDoIdx")
+					pos := task.Pos
+					logger := NewTimingLoger(SysLogger, pos, task.Msg).With(zap.Any("w", idx + 1))
+					tt := createTrainTask(TODOCTX, logger, task.Msg, task.W)
+					tt.Init("head")
+					tt.DCall(pos.X, pos.Y)
+					task.Call()
+				}()
+            }
+
+            SysLogger.Warn("CDoGScenWorldTask_Stop", zap.Any("name", s.Name))
+        }(i)
+    }
+}
+func init(){
+	WORLD_STATE_CODE = "5"
+    RegisterWorldRouter(0, GSceneWorldRouter)
+    RegisterDoWorldTask(int(SceneMode_Guild), GSceneDoWorldTask)
+	RegisterDoWorldTask(int(SceneMode_NewBie), NewBieSceneDoWorldTask)
+}
+
+/////
\ No newline at end of file
